### PR TITLE
Implement /api/ledger/addresses

### PR DESCRIPTION
**Pull request by gaborh-da**

### CODE REVIEW

File: gui-backend/src/main/java/com/rln/gui/backend/implementation/balanceManagement/data/AccountInfo.java [hunk 1]
```diff
@@ -12,12 +12,14 @@
...
     private String provider;
+    private String owner;
     private String iban;
     private String assetCode;
 
   public AccountInfo(Balance balance) {
     provider = balance.provider;
     iban = balance.iban;
     assetCode = balance.currency;
+    owner = balance.owner.party.orElse(balance.owner.name);
```
Comment (mjuchli-da) [inline gui-backend/src/main/java/com/rln/gui/backend/implementation/balanceManagement/data/AccountInfo.java:23]: I suggest adding `Objects.requireNonNull(balance.owner)` before this.

File: gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/AutoapproveApiImpl.java [hunk 2]
```diff
@@ -29,40 +30,36 @@
 public class AutoapproveApiImpl {
 
   private static final Filter AUTO_APPROVE_MARKER_FILTER = new InclusiveFilter(
-    Set.of(AutoApproveTransferProposalMarker.TEMPLATE_ID), Map.of());
+      Set.of(AutoApproveTransferProposalMarker.TEMPLATE_ID), Map.of());
```
Comment (mjuchli-da) [inline gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/AutoapproveApiImpl.java:33]: Let's stick to 4 spaces in Java. https://google.github.io/styleguide/javaguide.html#s4.5.2-line-wrapping-indent

File: gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/AutoapproveApiImpl.java [hunk 3]
```diff
@@ -90,6 +90,14 @@ public List<LedgerAddressDTO> get() {
     return result;
   }
 
+  private Long getSetlId(String damlPartyId) {
```
Comment (mjuchli-da) [inline gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/AutoapproveApiImpl.java:93]: I suggest converting this to `Map<String, Long> createPartyToSetlId()` and then on L74 (prior to for-loop) `var partyToSetlId = createPartyToSetlId()`. Furthermore, the construction of such a map could take place in `SetlPartySupplier` and expose a method `Optional<Long> getSetlId(String party)`.

**Review by mjuchli-da — CHANGES_REQUESTED** [review]

File: gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/AutoapproveApiImpl.java (hunk 2 above)
Comment (gaborh-da) [reply gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/AutoapproveApiImpl.java:33]: With Judy, we decided to use that formatter and set it up in IntelliJ, but sometimes (at least for me) it does not do its job.

File: gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/AutoapproveApiImpl.java (hunk 3 above)
Comment (gaborh-da) [reply gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/AutoapproveApiImpl.java:93]: I exposed a method that simply returns Long (as we are fine to put null into the result DTO). Is that okay?

File: gui-backend/src/main/java/com/rln/gui/backend/implementation/balanceManagement/data/AccountInfo.java (hunk 1 above)
Comment (gaborh-da) [reply gui-backend/src/main/java/com/rln/gui/backend/implementation/balanceManagement/data/AccountInfo.java:23]: Added. There might be other places where we are missing such a call.

File: gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/AutoapproveApiImpl.java (hunk 2 above)
Comment (gaborh-da) [reply gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/AutoapproveApiImpl.java:33]: I was wrong. That very line was formatted by the Google Formatter plugin following the Google Style (as is, without changes).
Screenshot from 2023-01-27 08-53-39

Comment (gaborh-da) [reply gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/AutoapproveApiImpl.java:33]: Because being unused, I just removed the line.

**Review by mjuchli-da — APPROVED** [review]

**Review by mjuchli-da — APPROVED** [review]

**Pull request merged**